### DISCUSSION TITLE
How can i access previous State when updating?

### DISCUSSION THREAD

**Discussion by MariuzM** [pmndrs/zustand#891] [Q&A]
Sorry i know this is not SO and these question should be related there but maybe this is particular case that developer would know only? I would like to use `userStateSet((prevState) => ({ ...prevState, lastname: 'last' }))` this will allow me to access previous state and update the rest of the object without losing properties. 

2 solutions i have but i don't like them:
```ts
useStorePreviousState.setState(({ user }) => {
  return { user: { ...user, lastname: 'last' } }
})
```
and 
```ts
userStateSet({ ...useStorePreviousState.getState().user, lastname: 'last' })
```

This would be my preferred solution 
```ts
userStateSet((prevState) => ({ ...prevState, lastname: 'last' }))
```

Full example:
https://codesandbox.io/s/quirky-banzai-4ced06?file=/src/App.tsx

**Comment by dai-shi** [comment]
There's no magic. I think you can define your own helper function.
```js
const userStateSet = (fn) => useStorePreviousState.setState(({ user }) => ({ user: fn(user) }))
```
